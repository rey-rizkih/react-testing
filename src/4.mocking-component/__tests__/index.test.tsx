/* eslint-disable no-console */
import React from "react";

import { render } from "@testing-library/react";

import Component from "..";

jest.mock("../BigComponent", () => () => (
  <p data-testid="mock-big-component" />
));

describe("Mocking", () => {
  test("mock component with jest mock", () => {
    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });
});
