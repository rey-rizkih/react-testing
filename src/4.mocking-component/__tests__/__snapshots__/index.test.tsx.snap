// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mocking mock component with jest mock 1`] = `
<div>
  <div>
    <p
      data-testid="mock-big-component"
    />
    component
  </div>
</div>
`;
